diode/otlp_client.go

```diff
@@ -0,0 +1,336 @@
+package diode
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"runtime"
+	"strings"
+	"time"
+
+	logsservicepb "go.opentelemetry.io/proto/otlp/collector/logs/v1"
+	commonpb "go.opentelemetry.io/proto/otlp/common/v1"
+	logspb "go.opentelemetry.io/proto/otlp/logs/v1"
+	resourcepb "go.opentelemetry.io/proto/otlp/resource/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/encoding/protojson"
+
+	"github.com/netboxlabs/diode-sdk-go/diode/v1/diodepb"
+)
+
+const (
+	otlpClientName        = "diode-sdk-go-otlp"
+	defaultOtlpTimeout    = 10 * time.Second
+	otlpExportErrorPrefix = "OTLP export failed"
+)
+
+// OTLPClientOption configures the OTLPClient.
+type OTLPClientOption func(*OTLPClient) error
+
+// OTLPClient implements Client and exports entities as OTLP log records.
+type OTLPClient struct {
+	appName    string
+	appVersion string
+
+	timeout time.Duration
+	stream  string
+
+	certFile string
+
+	metadata metadata.MD
+
+	conn   *grpc.ClientConn
+	client logsservicepb.LogsServiceClient
+
+	sdkName    string
+	sdkVersion string
+
+	platform  string
+	goVersion string
+}
+
+// WithOtlpTimeout overrides the default export timeout.
+func WithOtlpTimeout(timeout time.Duration) OTLPClientOption {
+	return func(c *OTLPClient) error {
+		if timeout <= 0 {
+			return fmt.Errorf("OTLP timeout must be greater than zero")
+		}
+		c.timeout = timeout
+		return nil
+	}
+}
+
+// WithOtlpCertFile configures the certificate file to trust for secure endpoints.
+func WithOtlpCertFile(certFile string) OTLPClientOption {
+	return func(c *OTLPClient) error {
+		c.certFile = certFile
+		return nil
+	}
+}
+
+// WithOtlpStream overrides the default stream value associated with exported entities.
+func WithOtlpStream(stream string) OTLPClientOption {
+	return func(c *OTLPClient) error {
+		if strings.TrimSpace(stream) == "" {
+			return fmt.Errorf("OTLP stream must not be empty")
+		}
+		c.stream = stream
+		return nil
+	}
+}
+
+// NewOTLPClient creates a new OTLPClient that exports entities as OTLP log records.
+func NewOTLPClient(target, appName, appVersion string, opts ...OTLPClientOption) (Client, error) {
+	if target == "" {
+		return nil, fmt.Errorf("target is required")
+	}
+
+	if !strings.HasPrefix(target, "grpc://") && !strings.HasPrefix(target, "grpcs://") {
+		return nil, fmt.Errorf("OTLPClient target should start with grpc:// or grpcs://")
+	}
```

Comment:
Can we use similar matching like in normal client, which also supports http and https schemes?
https://github.com/netboxlabs/diode-sdk-go/blob/develop/diode/client.go#L65

Reply:
we actually call it bellow in line 97. so this part of the code is not needed. I will remove it